### PR TITLE
Added remove and disable common hook

### LINKED ISSUE #31: Provide a collection of common hooks
Already suggested by @marshallswain in https://github.com/feathersjs/feathers-commons/issues/13 `feathers-hooks` should provide a small collection of common hooks. I am currently thinking of:

- Remove field
- Timestamp (`updatedAt`, `createdAt` etc.)
- Disabling methods (which should take care of https://github.com/feathersjs/feathers/issues/106)

Any others?

### PR DESCRIPTION
**Pull request by marshallswain**
Check README.md for usage.  Basically, the common hooks have to be imported by doing
```js
import hooks from 'feathers-hooks/hooks';
```

Closes #31 and closes https://github.com/feathersjs/feathers/issues/106

### CODE REVIEW

**Comment by ekryski** [comment]
I think the PR is good but I'm not totally convinced that we need/should provide common hooks this way. Can we wait until the example is finished? Some more common hooks may come out of that but I have a hunch that most of the hooks will live in each service (ie. auth, mandrill, stripe, etc.)

**Comment by marshallswain** [comment]
Yeah, it can wait.  Since I missed that conversation, I'm not sure how you're planning on implementing the API.  We'll probably be able to reuse hook and tests within what you have in mind, though.

**Comment by daffl** [comment]
If we don't add it here as a common hook, the solution for https://github.com/feathersjs/feathers/issues/106 will have become its own plugin.

File: src/hooks/remove-fields.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  fieldsToRemove.forEach(field => {
+    if (typeof field !== 'string') {
+      throw new Error('"removeFields" hook: field names must be strings.');
+    }
+  });
+
+  return function (hook, next) {
+    // Skip removing fields if it's flagged as internal.
+    if (!hook.params.internal) {
```
Comment (marshallswain) [inline src/hooks/remove-fields.js]: @daffl @ekryski This check for the internal flag might need to change.  When I implemented this I needed a way to keep the attribute on the server, but not out to public.  Have we decided on how the internal/external flagging API will work?

Comment (ekryski) [inline src/hooks/remove-fields.js]: I'm wondering if we should be more explicit. `hook.params.internal` makes sense to us but probably doesn't make sense to anyone new to Feathers. Possibly it should be called `isPublic`, `isPrivate`? Just spit ballin' here....

Comment (daffl) [inline src/hooks/remove-fields.js]: The v2 providers set the `prams.provider` property (to `rest`, `socketio` and `primus` respectively). You could just check for `hook.params.provider`.

Comment (ekryski) [inline src/hooks/remove-fields.js]: :+1: Ya I like that better.

Comment (marshallswain) [inline src/hooks/remove-fields.js]: Perfect.

**Pull request merged**